pennylane/_qubit_device.py

```diff
@@ -1459,7 +1459,7 @@ def circuit(x):
         if mp.obs is None and not isinstance(mp.mv, MeasurementValue):
             # convert samples and outcomes (if using) from arrays to str for dict keys
             samples = np.array([sample for sample in samples if not np.any(np.isnan(sample))])
-            samples = qml.math.cast_like(samples, qml.math.int8(0))
+            samples = qml.math.cast_like(samples, qml.math.int64(0))
```

Comment:
I'm a bit confused here @mudit2812. Is `samples` is a `(shots, len(wires))` array of zeros and ones? Why would $\\{0, 1\\}$ be incompatible with the range of `int8`?

Reply:
Earlier in the function, each sample (list of 0s and 1s), is converted to a decimal integer, so the shape then becomes `(shots,)`. Then we use f-strings to convert the integers to binary strings.